CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -131,7 +138,24 @@ namespace CefSharp
 
         void ClientAdapter::OnLoadingStateChange(CefRefPtr<CefBrowser> browser, bool isLoading, bool canGoBack, bool canGoForward)
         {
-            _browserControl->SetLoadingStateChange(canGoBack, canGoForward, isLoading);
+            if (browser->IsPopup())
+            {
+                auto browserId = browser->GetIdentifier();
+                IBrowser^ entry;
+                IPopupHandler^ handler = _browserControl->PopupHandler;
+
+                if (_popupBrowsers->TryGetValue(browserId, entry))
```

Comment:
Check `handler` for `null` first? Slightly more efficient when there's no `handler` assigned.